hypershift-operator/controllers/hostedcluster/internal/platform/gcp/gcp.go

```diff
@@ -0,0 +1,58 @@
+package gcp
```

Comment:
This implementation provides no value in this PR. Please lets drop it and narrow it down to e.g. only initial API.

A possible path forward would be incremental PRs for: 
- API
- cli generated infra
- hc, nodepool and CPO controllers
That would give a minimal gcp HC CRUD

Then PRs for  iterative cycles

Reply:
ok, ty for the review , I create seperate PRs as suggested